src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java

```diff
@@ -350,7 +351,12 @@ private RepositoryFunction getHandler(Rule rule) {
       // Upon an exceptional exit, the fetching of that repository is over as well.
       env.getListener().post(new RepositoryFetching(repositoryName, true));
       env.getListener().post(new RepositoryFailedEvent(repositoryName, e.getMessage()));
-      throw e;
+      env.getListener().handle(Event.error("Error fetching repository: " + e.getMessage()));
+
+      // Rewrap the underlying exception to signal callers not to re-report this error.
+      throw new RepositoryFunctionException(
```

Comment:
nit: I don't see why this is reusing RepositoryFunctionException. The convention is for each SkyFunction to have its own wrapper, that can narrow the exceptions it takes. Then RepositoryFunction doesn't need to know about any of this.

Reply:
I agree with your point. However, from reading [RepositoryDelegatorFunction](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java;l=183;drc=d4d9e28140dedbcc84e3486d06b373fbc2fbc51f), it seems to me that `RepositoryFunctionException` is being used as if it's `RepositoryDelegatorFunctionException`. We can have a separate PR for that refactoring.